packages/react-lowcode/src/codegen/generation/generators/detail/grommet-detail-generator.ts

```diff
@@ -1,99 +1,81 @@
+import { GeneratorHelper } from "../helper";
+import ReactIntlFormatter from "../../react-components/react-intl/intl-formatter";
 import { GrommetDetailComponents } from "../../../definition/grommet/detail";
-import TypescriptHelper from "../../code-generation/ts-helper";
-import { Formatter } from "../../../definition/context-types"
+import { DetailGenerator } from "../detail/detail-generator-factory";
 
 export default class GrommetDetailGenerator
-  extends DetailGeneratorBase
   implements DetailGenerator {
-    constructor(generationContext: GenerationContext, entity: Entity) {
-      super(generationContext, entity);
-  }
+  private readonly _helper: GeneratorHelper
+  private _imports: ts.ImportDeclaration[] = []
+  private _context: GenerationContext
+  private _entity: Entity
+  private _intlFormatter: ReactIntlFormatter
 
+  constructor(generationContext: GenerationContext, entity: Entity) {
+    this._helper = new GeneratorHelper(generationContext, entity)
+    this._context = generationContext
+    this._entity = entity
+    this._intlFormatter = new ReactIntlFormatter(generationContext, this._imports)
+  }
   getDetailDefinition(): DetailComponentDefinitionBase {
     return GrommetDetailComponents;
   }
 
   generateDetailComponent(): PageComponent {
     var statements = this.createStatements();
-    var functionalComponent = createFunctionalComponent(
+
+    var functionalComponent = this.createConstFunction(
       "FormikComponent",
-      [],
       statements
     );
 
-    this._imports = [...this._imports, ...this.intlFormatter.getImports()];
+    this._imports = [...this._imports, ...this._intlFormatter.getImports()];
 
-    var uniqueImports = this.uniqueImports();
-    uniqueImports.push(
-      TypescriptHelper.createNameSpaceImport("React", "react")
+    var uniqueFileImports = uniqueImports(this._imports)
+    uniqueFileImports.push(
+      createNameSpaceImport("React", "react")
     );
-
-    return { functionDeclaration: functionalComponent, imports: uniqueImports };
-  }
-
-  private createStatements(): ts.Statement[] {
-    let statements = new Array<ts.Statement>();
-
-    if(this.context.formatter === Formatter.Intl){
-      statements.push(this.intlFormatter.getImperativeHook())
-    }
-
-    var formikComponentDefinition = this.prepareComponent(
-      this.getDetailDefinition().formik
-    );
-
-    var fnameFieldComponent = this.createFormikTextFieldElement(
-      "firstName",
-      "First Name"
+    uniqueFileImports.push(
+      createImportDeclaration(
+        "TextInput",
+        "grommet"
+      )
     );
-    var lnameFieldComponent = this.createFormikTextFieldElement(
-      "lastName",
-      "Last name"
+    uniqueFileImports.push(
+      createImportDeclaration("useFormik", "formik")
     );
-    var emailFieldComponent = this.createFormikTextFieldElement(
-      "email",
-      "Email"
+
+    uniqueFileImports.push(
```

Comment:
If it is unique then it could be `Set`. If it is not possible than change the name.